File: src/main/webapp/app/js/services/WordsService.js
angular.module('myApp').service('WordsService', function ($http, $log) {
    var prefix = '/ejs';

    this.saveWord = function(word){
        return $http({
            method: 'POST',
            data: word,
            url: prefix+'/api/words'
        });
    }

    this.getAllWords = function (){
        return $http({
            method: 'GET',
            url: prefix+'/api/words'
        });
    };

    this.getWordLevels = function (){
        return $http({
            method: 'GET',
            url: prefix+'/api/words/levels'
        });
    };
    this.getWordsForLevel = function (level){
        return $http({
            method: 'GET',
            url: prefix+'/api/words/'+level
        });
    };

    this.deleteWord = function(praise){
        return $http({
            method: 'DELETE',
            params: {
                id: praise.id
            },
            url: prefix+'/api/words'
        });
    }

});

File: src/main/java/com/ssoward/service/WordsService.java
package com.ssoward.service;

import com.ssoward.model.Word;

import java.util.List;

/**
 * Created by ssoward on 3/5/14.
 */

public interface WordsService {

    List<Word> getWords();

    List<Word> getWordsForLevel(String level);

    void saveWord(Word word);

    void deleteWord(Long id);
}
